chemprop/featurizers/molgraph/reaction.py

```diff
@@ -135,9 +134,7 @@ def __call__(
                 edge_index[0].extend([u, v])
                 edge_index[1].extend([v, u])
 
-                i += 2
-
-        E = np.array(E)
+        E = np.array(E) if E else np.empty((0, self.bond_fdim))
```

Comment:
implicit calls to `bool()` are best avoided unless you _know_ it's the behavior you want.

Reply:
good point - how about `E = np.array(E) if len(E) > 0 else ` ...  ?

Reply:
I'd have to assume that's the exact behavior you want to be testing, so makes sense to me